src/Microsoft.Net.Http.Headers/MediaTypeHeaderValue.cs

```diff
@@ -457,17 +485,17 @@ private static int GetMediaTypeExpressionLength(string input, int startIndex, ou
 
             // If there is no whitespace between <type> and <subtype> in <type>/<subtype> get the media type using
             // one Substring call. Otherwise get substrings for <type> and <subtype> and combine them.
-            var mediatTypeLength = current + subtypeLength - startIndex;
-            if (typeLength + subtypeLength + 1 == mediatTypeLength)
+            var mediaTypeLength = current + subtypeLength - startIndex;
+            if (typeLength + subtypeLength + 1 == mediaTypeLength)
             {
-                mediaType = input.Substring(startIndex, mediatTypeLength);
+                mediaType = input.Substring(startIndex, mediaTypeLength);
             }
             else
             {
                 mediaType = input.Substring(startIndex, typeLength) + "/" + input.Substring(current, subtypeLength);
             }
```

Comment:
I didn't try to remove these substrings as part of this change. If we wanted to re-engineer all of this code around `StringSegment` we could get rid of them too.